mission/landmark_server/src/landmark_server.cpp

```diff
@@ -0,0 +1,294 @@
+#include "landmark_server/landmark_server.hpp"
+
+using std::placeholders::_1, std::placeholders::_2;
+using LandmarkArray = vortex_msgs::msg::LandmarkArray;
+using Action = vortex_msgs::action::FilteredLandmarks;
+
+namespace landmark_server {
+
+LandmarkServerNode::LandmarkServerNode(const rclcpp::NodeOptions &options)
+    : Node("landmark_server_node", options) {
+
+  // Define the quality of service profile for publisher and subscriber
+  rmw_qos_profile_t qos_profile = rmw_qos_profile_sensor_data;
+  qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+  auto qos = rclcpp::QoS(rclcpp::QoSInitialization(qos_profile.history, 1),
+                         qos_profile);
+
+  storedLandmarks_ = std::make_shared<LandmarkArray>();
+
+  landmark_sub_ = this->create_subscription<LandmarkArray>(
+      "target_tracking/landmarks", qos,
+      std::bind(&LandmarkServerNode::landmarksRecievedCallback, this, _1));
+
+  landmarkPublisher_ =
+      this->create_publisher<LandmarkArray>("landmarks_out", qos);
+
+  posePublisher_ = this->create_publisher<geometry_msgs::msg::PoseArray>(
+      "landmark_poses_out", qos);
+
+  tf2_buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
+  tf2_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf2_buffer_);
+
+  declare_parameter<std::string>("target_frame", "base_link");
+
+  // Create the act.
+  action_server_ = rclcpp_action::create_server<Action>(
+      this, "landmark_filter",
+      std::bind(&LandmarkServerNode::handle_goal, this, _1, _2),
+      std::bind(&LandmarkServerNode::handle_cancel, this, _1),
+      std::bind(&LandmarkServerNode::handle_accepted, this, _1));
+}
+
+void LandmarkServerNode::landmarksRecievedCallback(
+    const LandmarkArray::SharedPtr msg) {
+  if (msg->landmarks.empty()) {
+    RCLCPP_WARN_STREAM_THROTTLE(this->get_logger(), *this->get_clock(), 5000,
+                                "Received empty landmark array");
+    return;
+  }
+  const auto &pose = msg->landmarks[0].odom.pose.pose;
+  for (const auto &landmark : msg->landmarks) {
+    // RCLCPP_INFO(this->get_logger(), "Landmarks received");
+
+    if (landmark.action == 0) {
```

Comment:
can you get the landmark action enum from the message type? If so, change this to switch-case instead of else-if